src/time/Waveform.hpp

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include <Eigen/Core>
+#include "logging/Logger.hpp"
+
+namespace precice {
+
+namespace testing {
+// Forward declaration to friend the boost test struct
+class WaveformFixture;
+} // namespace testing
+
+namespace time {
+
+/**
+ * @brief Stores data samples in time and allows to perform interpolation to obtain new values.
+ */
+class Waveform {
+  friend class testing::WaveformFixture; // Make the fixture friend of this class
+public:
+  /**
+   * @brief Waveform object which stores values of current and past time windows for performing interpolation. 
+   *
+   * Storage still needs to be initialized with Waveform::initialize, before the Waveform can be used.
+   *
+   * @param interpolationOrder Defines the maximum interpolation order supported by this Waveform and reserves storage correspondingly
```

Comment:
```suggestion
   * @param interpolationOrder Defines the interpolation order supported by this Waveform and reserves storage correspondingly
```
I understand what you mean here, but others further away might not. Not the right place to drop the this information.

Reply:
I don't really agree here. Currently a `Waveform` is an object which allows the user to store data and interpolate based on the data (This is also explained in the documentation of the class). Just looking at this class I think that this explanation is appropriate. Or what exactly do you mean? Where would you give this information?

Reply:
My comment is only about the word "maximum" here.
With "maximum" you mean that the actual order might be smaller in the first time windows if we still have to build up some higher-order interpolant, right?
This information should be given somewhere explicitly (if true) and not hidden in the implicit "maximum".
Understandable what I mean?

Reply:
I added more documentation on this for the class in 8fc90224123047c8e1193e9799c23ee1ebfa65c2.